src/bindings/python/src/openvino/utils/decorators.py

```diff
@@ -76,16 +76,16 @@ def __init__(self, name: str):
         self.typemap: Dict[tuple, Callable] = {}
 
     # Checks if actual_type is a subclass of any type in the union
-    def matches_union(self, union_type, actual_type) -> bool:  # type: ignore
+    def matches_union(self, union_type: Any, actual_type: Type) -> bool:
         for type_arg in get_args(union_type):
-            if isinstance(type_arg, type) and issubclass(actual_type, type_arg):
+            if isinstance(type_arg, type) and isinstance(actual_type, type) and issubclass(actual_type, type_arg):
                 return True
             elif get_origin(type_arg) == list:
-                if issubclass(actual_type, list):
+                if isinstance(actual_type, type) and issubclass(actual_type, list):
                     return True
```

Comment:
it could be one condition, elif makes no sense here, imho
@akuporos what's your opinion?